src/main/java/upeldev/com/github/upel3/controllers/ActivityController.java

```diff
@@ -69,11 +69,14 @@ public String activity(Model model, Principal principal, HttpServletRequest requ
 
             model.addAttribute("activity", activity);
 
+            Course course = activity.getCourse();
+            model.addAttribute("course", course);
+
             int passValue = activity.getPassValue();
             double maxPoints = activity.getMaxPoints();
             double points = 0;
             List<Grade> grades;
-            if(currentUser.getRoles().contains(Role.STUDENT)){
+            if(currentUser.getRoles().contains(Role.STUDENT) || (currentUser.getRoles().contains(Role.LECTURER) && !currentUser.getCoursesLectured().contains(course))){
```

Comment:
checking whether coursesEnrolledIn contains current course be better than those convoluted conditions

Reply:
Yeah, that's way simpler.